pkg/requester/jobtransform/docker_image_digest.go

```diff
@@ -0,0 +1,46 @@
+package jobtransform
+
+import (
+	"context"
+
+	"github.com/bacalhau-project/bacalhau/pkg/docker"
+	"github.com/bacalhau-project/bacalhau/pkg/model"
+	"github.com/rs/zerolog/log"
+)
+
+func DockerImageDigest() Transformer {
+	return func(ctx context.Context, j *model.Job) (modified bool, err error) {
+		client, err := docker.NewDockerClient()
+		if err != nil || !client.IsInstalled(ctx) {
+			// Return a noop if docker is not installed as it means we
+			// won't be able to find digests for images in the requester
+			return false, nil
+		}
```

Comment:
Is it worth doing this stuff up front and forming a closure over the client? The `IsInstalled` check makes a req against the Docker daemon – is it bad to do this in the job submission hot path?

Reply:
It was originally like that, but there was no context available and it got called out in earlier review.  Happy to move it back into the outer function so we can close over it instead (and return a nop function earlier) - but we will have to add a pointless context as there isn't one available in the caller and I don't want to pointlessly add contexts everywhere.